Tools/clinic/clinic.py

```diff
@@ -4790,8 +4788,6 @@ def state_modulename_name(self, line: str) -> None:
         if not return_converter:
             return_converter = CReturnConverter()
 
-        if not module:
-            fail("Undefined module used in declaration of {full_name.strip()!r}.")
```

Comment:
`module` is always either an instance of `Module` or an instance of `Clinic`, and both are always truthy